### PR TITLE
Bump dugite to 2.5.2

### PR DESCRIPTION
**Pull request by DeeDeeG**
### Description of the Change

- Bump dugite to 2.5.2
- Bump whats-my-line to a dedicated branch where the only new change is bumping dugite to 2.5.2
  - (avoiding the superstring bump on whats-my-line `master`, as I'm not set up to sync the superstring bump across all three packages that depend on it at once just yet)

### Screenshot or Gif

N/A

### Applicable Issues

See also: https://github.com/pulsar-edit/whats-my-line/pull/8
Follows up on: https://github.com/pulsar-edit/github/pull/36

### CODE REVIEW

**Review by confused-Techie — APPROVED** [review]
This PR does in fact just do the bump as expected.

Although without tests hard to tell exactly where we are. But fingers crossed we are good to go

**Comment by DeeDeeG** [comment]
Looking at the CI, the baseline state of things isn't super great but I don't see anything that looks truly like a new issue as of this PR.

- Number of failing tests is **identical** on the Ubuntu-20.04 runs of both latest `master` commit and for this PR (18).
- Number of fails is **up 19 --> 22** for the ubuntu-latest test runs, but I didn't see any categorically new-looking ones. I think it's just flakes in the number of sub-tests for problematic test areas (`integration: file patches` and `WorkerManager`).
  - Reading the most into this, it could be a tiny perf regression, but perf is so variable on CI machines for GitHub Actions I would _not_ read this much into it, personally.
- macOS and Windows test runs, on `master` branch even before this PR, (consequently also for this PR of course) are broken, failing to even install the dependencies (such as for Python 3.12 issues, etc.)
  - These are too badly maintained to even meaningfully look at at the moment :/ They need a fix when someone gets the chance.

**Comment by DeeDeeG** [comment]
I'll merge this and ask @Daeraxa if this works on their ARM Linux machines before doing a PR to core?

(How to test: `ppm install https://github.com/pulsar-edit/github` after this merges, or clone this repo and `ppm install` + `ppm link`).

**Pull request merged**
